lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -493,23 +577,53 @@ def class_for_type(pe_type)
       @pe_type_class_hash[pe_type]
     end
 
-    # Do the pe's stored attributes match the extra_attributes provided in the query
+    # Check if the PolicyElement's extra_attributes column includes the passed
+    # key and value
     def pe_matches_extra_attributes?(policy_element, key, value, ignore_case)
       if policy_element.store_attributes
         attr_value = policy_element.extra_attributes_hash[key]
 
-        if ignore_case_applies?(ignore_case, key) && attr_value.is_a?(String) && value.is_a?(String)
-          attr_value.downcase == value.downcase
+        if value.is_a?(Array)
+          value.any? do |v|
+            extra_attribute_match(
+              ignore_case: ignore_case,
+              key: key,
+              value_1: attr_value,
+              value_2: v
+            )
+          end
         else
-          attr_value == value
+          extra_attribute_match(
+            ignore_case: ignore_case,
+            key: key,
+            value_1: attr_value,
+            value_2: value
+          )
         end
       end
     end
 
+    def extra_attribute_match(ignore_case:, key:, value_1:, value_2:)
+      if ignore_case_applies?(ignore_case, key) && value_1.is_a?(String) && value_2.is_a?(String)
+        value_1.downcase == value_2.downcase
+      else
+        value_1 == value_2
+      end
+    end
+
     # Allow ignore_case to be a boolean, string, symbol, or array of symbols or strings
     def ignore_case_applies?(ignore_case, key)
       return false if key == 'policy_machine_uuid'
-      ignore_case == true || ignore_case.to_s == key || ( ignore_case.respond_to?(:any?) && ignore_case.any? { |k| k.to_s == key } )
+
+      return true if ignore_case == true
```

Comment:
Can we just do the following?
```ruby
return true if ignore_case
```

Reply:
Ah, just saw the lines below. Retracted.

Reply:
If only the world were that simple!